### PR TITLE
Change nginx instrumentation into command line

### PR DESCRIPTION
**Pull request by dexter0195**
**Description:**
Change nginx instrumentation into command line

**Link to tracking Issue(s):** 
https://github.com/open-telemetry/opentelemetry-operator/issues/2676

- Resolves: #2676

**Testing:**

**Documentation:**

### CODE REVIEW

**Comment by dexter0195** [comment]
Hey 😃 
Can you help me understand why the e2e are all failing at the step prepare-e2e please 🙏 ?

**Comment by TylerHelmuth** [comment]
The instrumentation was off by default, so you'll need to update some tests to re-enable it.  You can look at https://github.com/open-telemetry/opentelemetry-operator/pull/2750 as an example.

File: main.go [hunk 1]
```diff
@@ -140,6 +141,7 @@ func main() {
 	pflag.BoolVar(&enableApacheHttpdInstrumentation, constants.FlagApacheHttpd, true, "Controls whether the operator supports Apache HTTPD auto-instrumentation")
 	pflag.BoolVar(&enableDotNetInstrumentation, constants.FlagDotNet, true, "Controls whether the operator supports dotnet auto-instrumentation")
 	pflag.BoolVar(&enablePythonInstrumentation, constants.FlagPython, true, "Controls whether the operator supports python auto-instrumentation")
+	pflag.BoolVar(&enableNginxInstrumentation, constants.FlagNginx, true, "Controls whether the operator supports nginx auto-instrumentation")
```
Comment (TylerHelmuth) [inline main.go:144]: This should be disabled by default.

**Comment by dexter0195** [comment]
> The instrumentation was off by default, so you'll need to update some tests to re-enable it. You can look at #2750 as an example.

Hey @TylerHelmuth thank you for the advice, I had a look at the pull request you said and performed the required changes.

However the tests were still failing so I found that we were activating the feature gate in config/manager/manager.yaml , which after its removal was no longer valid. So I removed it and no the e2e tests are passing ✅ 

Please tell me if the solution I found was the right one or if there is something else I should do 🙏 
Thank you

**Comment by swiatekm-sumo** [comment]
It looks like we have both the Go and the Nginx autoinstrumentations enabled in our release manifests, I believe unintentionally? If I recall correctly, the intent was to just enable them in E2E tests, right @TylerHelmuth ? In that case, we should probably just accept this state of affairs and set the flags themselves to be enabled by default. WDYT @jaronoff97 @pavolloffay ?

To be clear, I think this PR can be merged as-is, it's not changing the current behaviour.

**Comment by TylerHelmuth** [comment]
> If I recall correctly, the intent was to just enable them in E2E tests

Correct.

**Review by swiatekm-sumo — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Comment by dexter0195** [comment]
Hey 😃 
thank you for the review and your comments ! Do you think we can merge this ?

File: .chloggen/autoinstrumentation-nginx-cli-flag.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. operator, target allocator, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: change nginx instrumentation feature gate into command line flag --enable-nginx-instrumentation
```
Comment (pavolloffay) [inline .chloggen/autoinstrumentation-nginx-cli-flag.yaml:8]: Please mention here what is the feature gate flag that is being changed.

Comment (dexter0195) [reply .chloggen/autoinstrumentation-nginx-cli-flag.yaml]: Done ✅

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**